nginx/config.go

```diff
@@ -73,8 +73,9 @@ http {
 	# treated as if it were HTTPs.
 	map $http_x_forwarded_proto $tmp_access_scheme {
 		default $scheme;               # if X-Forwarded-Proto header is empty, $tmp_access_scheme will be the actual protocol used
-		"~^(.*, ?)?http$" "http";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value
-		"~^(.*, ?)?https$" "https";    # account for the possibility of a comma-delimited X-Forwarded-Proto header value
+		"~^(.*, ?)?https(,.* ?)?$" "https";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value to see if https is present
+		"~^(.*, ?)?http(,.* ?)?$" "http";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value to see if http is present
```

Comment:
I had to put a lot of though into this and in the end, I'm not entirely sure about loosening up these two regexes like this.

Determining the access scheme is an exercise in determining when a request should be treated as an HTTPS request even though SSL offloading might already have occurred.

Before, the access scheme was considered HTTPS if (and only if) the potentially coma-delimited list of forwarded protocols _ended_ with HTTPS. That would be indicative of SSL having been offloaded by the most proximate proxy (usually your load balancer).

This change treats a request as HTTPS if _any_ intermediate proxy offloaded SSL (or claims to have done so).

Here's the critical question: If SSL were offloaded by some other proxy _before_ it reached your load balancer, should we really regard that request as secure? Since this means an unencrypted request may have traversed untrusted networks, I would argue no... and that's _without_ any malicious actors in the mix. If you assume there could be, there's no shortage of creative means by which I can think of abusing this.

Let me ask... how critical is loosening these regexes to fixing #244? Unless I am mistaken, we could do without these two changes. Thoughts?